File: Classes/OutputClasses/EngineClasses/AirSpike.py
from ...GeneralClasses.ControlSurface import ControlSurface


class AirSpike(ControlSurface):
    def __init__(self) -> None:
        super().__init__()
        self.min_position.set(0)
        self.change_rate.set(1)

    def deploy(self):
        if not self.move_to_target():
            return
            #print(f'\n{self.name.get()} is not in tolerance.')

    def retract(self):
        self.return_to_zero()

    def calculate_target(self, delta_p:int):
        tgt_position = self.current_position.get() + delta_p
        self.set_target_position(tgt_position)

File: C++ Implimentation/Classes/Engines/Engine/AirSpike/AirSpike.cpp
#include "AirSpike.h"

    AirSpike::AirSpike()
    : barometer() 
    {
    min_position.setValue(0);
    change_rate.setValue(1);
    }

void AirSpike::calculate_target(){
    int tgt_position = current_position.getValue() + barometer.getValue();
    set_target_position(tgt_position);
}

void AirSpike::deploy(){
    if (move_to_target().getNotValue()){
        return;
        //#print(f'\n{self.name.get()} is not in tolerance.')
        }
}

void AirSpike::retract(){
    return_to_zero();
}

void AirSpike::run(){
    calculate_target();
    deploy();
}
